src/ArgentPonyWarcraftClient/Models/GameDataApi/Covenant/RenownReward.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Text.Json.Serialization;
+
+namespace ArgentPonyWarcraftClient
+{
+    /// <summary>
+    /// A covenant renown reward.
+    /// </summary>
+    public class RenownReward
+    {
+        /// <summary>
+        /// Gets the renown level that the reward is earned at.
+        /// </summary>
+        [JsonPropertyName("level")]
+        public int Level { get; set; }
+
+        /// <summary>
+        /// Gets the reward reference.
+        /// </summary>
+        [JsonPropertyName("reward")]
+        public RenownRewardReference Reward { get; set; }
```

Comment:
Yeah this one I struggled a little bit with, because the document model for a reward looks an *awful* lot like a `*Reference` model we use for other things. But at the same time, there aren't any other APIs to get a `RenownReward` from Blizzard, so maybe it's not necessarily a reference? Best alternative I can think of at the moment is a `RenownRewardReference`

Reply:
The rewards have `key` properties with `href` values like `https://us.api.blizzard.com/data/wow/covenant/reward/321?namespace=static-9.0.2_36532-us`.  Even though that is not an advertised part of the Covenant API, it looks like it's intended to be a real thing at some point.  Maybe it becomes a `CovenantRewardReference` based on the URI structure.